developer_docs/provisioning_setup.asciidoc

```diff
@@ -0,0 +1,258 @@
+[[api-v2-development]]
+= Provisioning setup
+:toc: right
+:toclevels: 5
+
+Guide how to setup provisioning on Fedora with Libvirt.
+
+[[tftp-configuration]]
+== TFTP configuration
+
+[source, bash]
+....
+dnf install -y syslinux
+mkdir -p /var/lib/tftpboot
+mkdir -p /var/lib/tftpboot/{boot,grub,grub2,pxelinux.cfg}
+cp /usr/share/syslinux/{pxelinux.0,menu.c32,chain.c32,ldlinux.c32,libcom32.c32,libutil.c32} /var/lib/tftpboot
+
+# Change file owner to your account
+chown your-account:your-group -R /var/lib/tftpboot
+....
+
+[[libvirt-configuration]]
+== Libvirt configuration
+Install Libvirt
+[source, bash]
+....
+dnf install @virtualization
+....
+
+Check that `libvirtd` is running
+[source, bash]
+....
+systemctl start libvirtd
+systemctl enable libvirtd
+systemctl status libvirtd
+....
+
+Edit Libvirt `default` network
+[source, bash]
+....
+# virsh net-edit default
+
+<network>
+  <name>default</name>
+  <uuid>your-uuid</uuid>
+  <forward mode='nat'/>
+  <bridge name='virbr0' stp='on' delay='0'/>
+  <mac address='your-mac'/>
+  <domain name='local.loc' localOnly='yes'/>
+
+  <dns>
+    <host ip='192.168.122.1'>
+      <hostname>foreman.local.loc</hostname>
+    </host>
+  </dns>
+
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <tftp root='/var/lib/tftpboot'/>
+    <dhcp>
+      <range start='192.168.122.2' end='192.168.122.254'/>
+      <bootp file='pxelinux.0'/>
+    </dhcp>
+  </ip>
+</network>
+....
+
+Reload network configuration
+[source, bash]
+....
+virsh net-destroy default
+virsh net-start default
+....
+
+[[network-configuration]]
+== Network configuration
+Configure `dnsmasq` for Libvirt
```

Comment:
Personally I use https://gist.github.com/ekohl/e487422d4aaad8ab176c19ec713b700a which sets up systemd-resolved using a hook libvirt, which reads the network config. I like it a lot and it's very little work to maintain. It means you can easily define multiple networks with different domains and it just does the right thing.